model/network/registry.yaml

```diff
@@ -236,3 +236,46 @@ groups:
         type: string
         brief: 'The network interface name.'
         examples: [ 'lo', 'eth0' ]
+      - id: network.connection.state
+        type:
+          members:
+            - id: close
```

Comment:
I think this is coming from https://datatracker.ietf.org/doc/html/rfc9293, can we 
- review values and adjust them to match the RFC? E.g. this one should be `closed`
- add link to the RFC in the brief?

Reply:
I did some digging, looking into where those are used/defined. It seems that they are defined in `hostmetrics` receiver:
1. https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6f3ff4a37efe4591f728250d4ea9e4ad6672fb2b/receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_others.go#L9-L20 
2. https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c4bee406f5db233a2bc7f0c69185764fea9fb033/receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_linux.go#L18-L29

Used at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c4bee406f5db233a2bc7f0c69185764fea9fb033/receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper.go#L178-L188

These match with what `gopsutil` provides:
1. https://github.com/shirou/gopsutil/blob/62a52f910bd9ad7d48a7416f5c2899386d50cb37/net/net_linux.go#L308
2. https://github.com/shirou/gopsutil/blob/62a52f910bd9ad7d48a7416f5c2899386d50cb37/net/net_windows.go#L565

Moreover, these seem to link to the OS specific definitions:
1. https://github.com/torvalds/linux/blob/cdd30ebb1b9f36159d66f088b61aee264e649d7a/include/net/tcp_states.h#L19
2. https://learn.microsoft.com/en-us/windows/win32/api/tcpmib/ns-tcpmib-mib_tcprow2?redirectedfrom=MSDN#members

So it seems that it is `CLOSED` for Windows while `CLOSE` for Linux. Should we allow both or should we fix the receiver's implementation to correctly map to RFC's values?

@braydonk @dmitryax what are your thoughts since you are the code-owners of the receiver?

Reply:
I defer to system semconv group to discuss and decide. I have a preference to stick to RFC definition (which is cross-platform) and allow OS-specific values. In case of `close`/`closed`, the method that translates linux enum number into otel string value can take care of reporting the standard RFC state name.

Reply:
Discussed in System Metrics WG today. We agree on having the SemConv following what the (cross-platform) RFC defines and then the receiver to properly do the mappings accordingly for each platform/implementation.

I will update this PR accordingly.